rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1824,7 +1830,81 @@ func Create_TerminationInstruction: <"Creates the relevant primitive instruction
 	set instruction -> quantityChange: <"Set primitive instruction to be only a quantity change instruction (with values set to zero).">
 		Create_QuantityChangeInstruction( [changePriceQuantity], QuantityChangeDirectionEnum->Replace, empty )
 
-// End of ICMA - Phase 2 Contribution
+func Create_PartialDeliveryPrimitiveInstruction: <"Creates the primitive instructions for a substitution of collateral by replacing the assetpayout of the trade.">
+	inputs:
+		tradeState TradeState (1..1) <"The original trade to be closed.">
+		effectiveDate AdjustableOrRelativeDate (1..1) <"The date to close and open a new position.">
+		terminationDate AdjustableOrRelativeDate (1..1) <"The new termination date.">
+		deliveredPriceQuantity PriceQuantity (1..*) <"The price and quantity of the remaining trade. ">
+		undeliveredPriceQuantity PriceQuantity (1..*) <"The price and quantity of the remaining trade. ">
+	output: 
+		instruction PrimitiveInstruction (1..1)
+
+	condition ContractualProduct: <"This function applies to contractual products only.">
+		tradeState -> trade -> tradableProduct -> product -> contractualProduct exists
+
+	set instruction -> split -> breakdown: <"Sets the first part of the split to be a termination instruction for the existing trade.">
```

Comment:
this is not a termination instruction, its a quantity change, will it close the original trade?
looking at the ERCC guide and ISLA best practices it looks like the recommended action in both markets is that for a partial delivery the original trade is closed out and a new one created with the undelivered quantity. For both it also says that the original trade does not _need_ to be closed, but in this instance our best practices align.